From 3930c446f497f2c6d7ad9b21145c4ede86d23050 Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Fri, 8 May 2020 16:14:46 +0800
Subject: [PATCH] driver: mxc: hdp: correct the code logical of making sure a
 value is power of 2

The code logical wants to make sure the max_ch_num has to
be power of 2. So instead the '*' with '&', and remove the inversion
mark '!'.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hdp/API_HDMI_RX_Audio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/hdp/API_HDMI_RX_Audio.c b/drivers/mxc/hdp/API_HDMI_RX_Audio.c
index 1d1442c731ff..aa10fbc5af02 100644
--- a/drivers/mxc/hdp/API_HDMI_RX_Audio.c
+++ b/drivers/mxc/hdp/API_HDMI_RX_Audio.c
@@ -107,7 +107,7 @@ CDN_API_STATUS CDN_API_RX_AudioAutoConfig(
 	if (max_ch_num < 2 || max_ch_num > 32)
 		return CDN_ERR;
 	/* Maximum number of channels has to be power of 2 */
-	else if (!(max_ch_num * (max_ch_num - 1)))
+	else if ((max_ch_num & (max_ch_num - 1)))
 		return CDN_ERR;
 	/* Each active port shall carry the same number of sub-channels */
 	else if (max_ch_num % i2s_ports_num)
-- 
2.17.1

